test/unit/modules/EmailRecoveryModule/constructor.t.sol

```diff
@@ -28,42 +29,59 @@ contract EmailRecoveryModule_constructor_Test is EmailRecoveryModuleBase {
         );
     }
 
-    function test_Constructor_RevertWhen_UnsafeOnInstallSelector() public {
-        vm.expectRevert(
-            abi.encodeWithSelector(
-                EmailRecoveryModule.InvalidSelector.selector, IModule.onInstall.selector
-            )
+    function test_Constructor_When_SafeAddOwnerSelector() public {
+        _skipIfNotSafeAccountType();
+        new EmailRecoveryModule(
+            address(verifier),
+            address(dkimRegistry),
+            address(emailAuthImpl),
+            address(emailRecoveryHandler),
+            validatorAddress,
+            ISafe.addOwnerWithThreshold.selector
         );
+    }
+
+    function test_Constructor_When_SafeRemoveOwnerSelector() public {
+        _skipIfNotSafeAccountType();
         new EmailRecoveryModule(
             address(verifier),
             address(dkimRegistry),
             address(emailAuthImpl),
             address(emailRecoveryHandler),
             validatorAddress,
-            IModule.onInstall.selector
+            ISafe.removeOwner.selector
         );
     }
 
-    function test_Constructor_RevertWhen_UnsafeOnUninstallSelector() public {
-        vm.expectRevert(
-            abi.encodeWithSelector(
-                EmailRecoveryModule.InvalidSelector.selector, IModule.onUninstall.selector
-            )
+    function test_Constructor_When_SafeSwapOwnerSelector() public {
+        _skipIfNotSafeAccountType();
+        new EmailRecoveryModule(
+            address(verifier),
+            address(dkimRegistry),
+            address(emailAuthImpl),
+            address(emailRecoveryHandler),
+            validatorAddress,
+            ISafe.swapOwner.selector
         );
+    }
+
+    function test_Constructor_When_SafeChangeThresholdSelector() public {
+        _skipIfNotSafeAccountType();
         new EmailRecoveryModule(
             address(verifier),
             address(dkimRegistry),
             address(emailAuthImpl),
             address(emailRecoveryHandler),
             validatorAddress,
-            IModule.onUninstall.selector
+            ISafe.changeThreshold.selector
         );
     }
 
-    function test_Constructor_RevertWhen_UnsafeExecuteSelector() public {
+    function test_Constructor_RevertWhen_SafeNotAllowedSelector() public {
```

Comment:
Nitpick:

Don't think this minor duplication matters, but fyi `test_Constructor_RevertWhen_SafeNotAllowedSelector` is just replicating the same logic as `test_Constructor_RevertWhen_UnsafeOnInstallSelector`, as the second function still gets called when the account type is Safe. Same for `test_AllowValidatorRecovery_RevertWhen_SafeNotAllowedSelector` in `allowValidatorRecovery.t.sol`

Happy to leave as is